### PR TITLE
Kernel cleanup related to associative words / pc words

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: lib/rwspcgrp.gi [hunk 1]
```diff
@@ -400,7 +416,7 @@ function( sc )
     fam!.8BitsType := NewType( fam, Is8BitsPcWordRep );
 
     # copy the assoc word type
-    for i  in [ AWP_FIRST_ENTRY .. AWP_FIRST_FREE-1 ]  do
+    for i in [ AWP_FIRST_ENTRY+1 .. AWP_LAST_ENTRY ] do
         fam!.8BitsType![i] := sc![SCP_DEFAULT_TYPE]![i];
```
Comment (fingolfin) [inline lib/rwspcgrp.gi]: Note that AWP_FIRST_ENTRY+1 = AWP_PURE_TYPE, and that entry is set directly after this loop. So the insertion of `+1` just ensures that we don't set that entry twice, and syncs us with the corresponding HPC-GAP code.

File: hpcgap/lib/rwspcgrp.gi [hunk 2]
```diff
@@ -400,7 +416,7 @@ function( sc )
     fam!.8BitsType := NewType( fam, Is8BitsPcWordRep );
 
     # copy the assoc word type
-    for i  in [ AWP_FIRST_ENTRY+1 .. AWP_FIRST_FREE-1 ]  do
```
Comment (hulpke) [inline hpcgap/lib/rwspcgrp.gi]: Is this fixing a bug, or did the size of the data type change?

Comment (fingolfin) [reply hpcgap/lib/rwspcgrp.gi]: I think our comments overlapped in time :-); here is what I wrote above:

Note that AWP_FIRST_ENTRY+1 = AWP_PURE_TYPE, and that entry is set directly after this loop. So the insertion of +1 just ensures that we don't set that entry twice, and syncs us with the corresponding HPC-GAP code.

Comment (hulpke) [reply hpcgap/lib/rwspcgrp.gi]: Thanks!

**Pull request merged**